src/main/java/net/minecraftforge/items/VanillaInventoryCodeHooks.java

```diff
@@ -50,7 +50,7 @@ public static Boolean extractHook(IHopper dest)
                 for (int j = 0; j < dest.getSizeInventory(); j++)
                 {
                     ItemStack destStack = dest.getStackInSlot(j);
-                    if (destStack == null || destStack.stackSize < destStack.getMaxStackSize() && ItemHandlerHelper.canItemStacksStack(extractItem, destStack))
+                    if (destStack == null || destStack.stackSize < destStack.getMaxStackSize() && destStack.getMaxStackSize() < dest.getInventoryStackLimit() && ItemHandlerHelper.canItemStacksStack(extractItem, destStack))
```

Comment:
Thank you for noticing that, @LexManos I am not sure how I made that mistake.  Fixed now.